index.js

```diff
@@ -1,20 +1,20 @@
 'use strict'
 
-const logger = require('./lib/logger')
+const Logger = require('./lib/logger.js')
 
-let __singleton
+let singleton
 
 const setupDefaultLogger = function(key, opts) {
-  if (!(__singleton instanceof logger.Logger)) {
-    __singleton = logger.createLogger(key, opts)
-  }
-  return __singleton
+  if (singleton) return singleton
+  singleton = new Logger(key, opts)
+  return singleton
+}
+
+function createLogger(key, options) {
+  return new Logger(key, options)
 }
 
 module.exports = {
-  cleanUpAll: logger.cleanUpAll
-, createLogger: logger.createLogger
-, flushAll: logger.flushAll
-, Logger: logger.Logger
+  createLogger
```

Comment:
would it be useful to export `Logger` from the entrypoint as well?

Reply:
The old version does, and we certainly could, but if we do that, why bother with the `createLogger` function at all?  And the singleton version as well.

Reply:
by "useful" the only thing I was thinking was for users extending the object for some reason, which may not a be a real use case. since we were exporting it before though, technically removing it becomes part of the breaking changes of this version?

Reply:
yes, that's part of the breaking change.  It either needs to be documented, or just exported like it was.  I guess I don't feel strongly, although as a user, I think it makes the `createLogger` function worthless. 🤷‍♂️

Reply:
i agree with you. part of me thinks we should instead just deprecate `createLogger` with a warning and suggest instantiating `Logger` yourself. 

deprecating `createLogger` undoubtedly has a larger impact than removing the exported `Logger` though, so I'd be fine with either documenting the removal or keeping it exported.